widgetssdk/src/main/java/com/glia/widgets/chat/ChatManager.kt

```diff
@@ -247,12 +254,57 @@ internal class ChatManager constructor(
         chatItems[index] = responseCard.asPlainText()
     }
 
+    @VisibleForTesting
+    fun mapMessageClicked(messageId: String, state: State): State {
+        val unsent = state.unsentItems.firstOrNull { it.messageId == messageId }
+
+        return if (unsent != null) {
+            state.apply {
+                onChatAction(Action.RemoveMessage(messageId))
+                sendUnsentMessagesUseCase(unsent, {
+                    onChatAction(Action.MessageSent(it))
+                }, {
+                    onChatAction(Action.MessageSendError(unsent))
+                })
+            }
+        } else {
+            state
+        }
+    }
+
     @VisibleForTesting
     fun addUnsentMessage(message: Unsent, state: State): State {
         state.unsentItems += message
         return state.apply {
             val index = if (chatItems.lastOrNull() is OperatorStatusItem.InQueue) chatItems.lastIndex else chatItems.lastIndex + 1
-            chatItems.add(index, message.chatMessage)
+            message.chatMessage?.let {
+                chatItems.add(index, it)
+            }
+        }
+    }
+
+    @VisibleForTesting
+    fun addErrorMessage(message: Unsent, state: State): State {
+        return state.apply {
+            unsentItems += message
+
+            val index = if (chatItems.lastOrNull() is OperatorStatusItem.InQueue) chatItems.lastIndex else chatItems.lastIndex + 1
```

Comment:
Honestly, I 'm not sure I understand this line. Maybe it makes sense to add a comment.

Reply:
Besides, does this mean that we add error message to the last chat item? What if the last message was delivered successfully, but some previous message delivery failed?